src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -361,6 +380,23 @@ CreateDistributedInsertSelectPlan(Query *originalQuery,
 }
 
 
+/*
+ * InsertSelectRouter is a helper function that returns true of the SELECT
+ * part of the INSERT .. SELECT query is a router query.
+ */
+static bool
+InsertSelectRouter(Query *originalQuery,
+				   PlannerRestrictionContext *plannerRestrictionContext)
+{
+	RangeTblEntry *subqueryRte = ExtractSelectRangeTableEntry(originalQuery);
+	DistributedPlan *distributedPlan = CreateRouterPlan(subqueryRte->subquery,
```

Comment:
CreateRouterPlan
	DeferErrorIfUnsupportedRouterPlannableSelectQuery
		AllDistributedRelationsInListColocated << Here shouldn't we consider target relation too for colocation?

Reply:
That is already done in DistributedInsertSelectSupported->AllDistributedRelationsInListColocated

But a good test to add for future in case of refactors

Reply:
we have bunch of such tests, you can see the output changes